### ISSUE TITLE
Allow to use clouds.yaml file to setup provider

### ISSUE THREAD

**Issue by mcanevet** [terraform-providers/terraform-provider-openstack#122]
OpenStack provides a standard path to store credentials in `~/.config/openstack/clouds.yaml` which is supported by `python-openstackclient` among others.
It would be great if the OpenStack provider could fetch credentials from this file.
For example:
```hcl
provider "openstack" {
  cloud = "my_cloud"
}
```
Juste like the aws providers allows:
```hcl
provider "aws" {
  profile = "my_profile"
}
```

**Comment by fatmcgav** [comment]
@mcanevet I like the idea... 

Looking at the AWS provider, it is leveraging functionality from AWS' go-sdk... 
So if we wanted to follow the same model, this should really be added as functionality to gophercloud before TF can implement; however not sure if there's appetite to add in gophercloud, in which case should we roll our own in tf-prov-openstack?

@jtopjian Thoughts?

**Comment by mcanevet** [comment]
@fatmcgav https://github.com/gophercloud/gophercloud/pull/44

**Comment by fatmcgav** [comment]
@mcanevet Heh, completely failed to spot that one...

**Comment by jtopjian** [comment]
I fully support the idea and I've been meaning to add this to `gophercloud/utils`. Once the Microversion issue is sorted out, I'll work on it.

**Comment by mcanevet** [comment]
@jtopjian did you have time to work on this?

**Comment by jtopjian** [comment]
@mcanevet No, I have not had time.

**Comment by jtopjian** [comment]
@mcanevet I forgot to mention that I begun working on this here:

https://github.com/gophercloud/utils/pull/2

Let me know if you have any questions or comments about the implementation.

**Comment by mcanevet** [comment]
Closed by #154
